conan/api/subapi/list.py

```diff
@@ -160,7 +163,20 @@ def filter_packages_profile(packages, profile, ref):
 
         return result
 
-    def select(self, pattern, package_query=None, remote=None, lru=None, profile=None):
+    def select(self, pattern: ListPattern, package_query=None, remote: Remote = None, lru=None, profile=None) -> PackagesList:
```

Comment:
I typed the args for the opaque objects here for better documentaion on the site, even if I'm still not a fan of wholly typing the API itself

Reply:
Remote only exposes its cache invalidation, and "no direct initialization" is documented, and ListPattern only exposes its constructor so that it can later be passed to the select method